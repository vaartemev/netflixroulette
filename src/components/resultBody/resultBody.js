import React, { useState, useEffect } from 'react';
import { MovieItem } from '../movieItem/movieItem';
import './resultBody.scss';

export const ResultBody = () => {
  const [movies, setMovies] = useState([]);

  useEffect(() => {
    fetch('http://react-cdp-api.herokuapp.com/movies')
      .then(res => res.json())
      .then(json => {
        const fetchedMovies = json.data.map(
          ({ title, poster_path, release_date, genres, id }) => (
            <MovieItem
              key={id}
              title={title}
              src={poster_path}
              year={release_date}
              genre={genres}
            />
          )
        );
        setMovies(fetchedMovies);
      });
  }, setMovies);
  return (
    <div className="content">
	  <div className="films-list">
	  {movies}
	  </div>
    </div>
  );
};