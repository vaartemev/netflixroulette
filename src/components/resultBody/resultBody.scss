.content {
	padding: 50px;
	background-color: #fdfdfd;
}
.films-list {
	display: grid;
	grid-template-columns: 1fr 1fr 1fr;
	justify-items: center
}

.no-films-found{
	width: 100%;
	height: 480px;
	text-align: center;
	line-height: 480px;
}