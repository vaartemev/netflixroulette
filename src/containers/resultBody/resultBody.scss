.content {
	background-color: #fdfdfd;
	height: 100%;
	&-not-found{
		width: 100%;
		height: 60vh;
		line-height: 60vh;
		text-align: center;
		&__text{
			font-size: 48px;
		}
	}
	&-films-list {
		padding: 50px;
		display: grid;
		grid-template-columns: 1fr 1fr 1fr;
		justify-items: center
	}
}

@media only screen and (max-width: 991px) {
	.content {
		&-films-list {
			grid-template-columns: 1fr 1fr;
		}
	}
}