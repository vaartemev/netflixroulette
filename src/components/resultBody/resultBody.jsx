import React from 'react';
import classNames from 'classnames';
import { useSelector, useDispatch } from 'react-redux';
import { MovieItem } from '../movieItem/movieItem';
import { NoFilmsFound } from '../noFilmsFound/noFilmsFound';
import './resultBody.scss';
import { getMovieDetailsById } from '../../actions';

export const ResultBody = () => {
  const movies = useSelector(state => state.movie.movies);
  const isFetching = useSelector(state => state.movie.isFetching);
  const dispatch = useDispatch();

  const resultClass = classNames({
    'no-films-found': movies.length === 0,
    'films-list': movies.length > 0
  });

  return (
    <div className="content">
      <div className={resultClass}>
        {isFetching ? (
          <h1>Загрузка...</h1>
        ) : movies.length === 0 ? (
          <NoFilmsFound />
        ) : (
          movies.map(({ id, title, poster_path, release_date, genres }) => (
            <MovieItem
              key={id}
              handleOnClick={() => dispatch(getMovieDetailsById(id))}
              title={title}
              src={poster_path}
              year={release_date}
              genre={genres}
            />
          ))
        )}
      </div>
    </div>
  );
};
