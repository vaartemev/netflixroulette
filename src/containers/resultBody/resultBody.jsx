import React, { useEffect } from 'react';
import classNames from 'classnames';
import { withRouter } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { Preloader, MovieItem, NoFilmsFound } from '../../components';

import { getMovieDetailsById, getMoviesBySearchQuery } from '../../actions';
import './resultBody.scss';

export const ResultBody = withRouter(({ match }) => {
  const { movies, isFetching } = useSelector(state => ({
    movies: state.movie.get('movies'),
    isFetching: state.movie.get('isFetching'),
  }));
  const dispatch = useDispatch();
  const yetLoader = (condition, then, otherwise) =>
    condition ? then : otherwise;

  const resultClass = classNames('content', {
    'content-not-found': movies.length === 0,
    'content-films-list': movies.length > 0,
  });

  const { searchValue } = match.params;
  useEffect(() => {
    dispatch(getMoviesBySearchQuery(searchValue));
  }, []);

  return (
    <div className={resultClass}>
      {movies.length === 0
        ? yetLoader(
            isFetching,
            <Preloader />,
            <NoFilmsFound className="content-not-found__text" />,
          )
        : movies.map(({ id, title, poster_path, release_date, genres }) => (
            <MovieItem
              key={id}
              id={id}
              handleOnClick={() => dispatch(getMovieDetailsById(id))}
              title={title}
              src={poster_path}
              year={release_date}
              genre={genres}
            />
          ))}
    </div>
  );
});
