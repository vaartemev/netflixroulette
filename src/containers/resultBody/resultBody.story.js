import React from 'react';
import { storiesOf } from '@storybook/react';
import { ResultBody } from './resultBody';

storiesOf('ResultBody', module).add('ResultBody', () => <ResultBody />);
