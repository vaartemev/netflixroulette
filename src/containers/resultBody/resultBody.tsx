import React, { useEffect } from 'react';
import classNames from 'classnames';
import { useRouter } from 'next/router';
import { useSelector, useDispatch } from 'react-redux';
import { Preloader, MovieItem, NoFilmsFound } from '../../components';

import { getMovieDetailsById, getMoviesBySearchQuery } from '../../actions';
import './resultBody.scss';

interface MoviesTypes {
  id: number;
  title: string;
  poster_path: string;
  release_date: string;
  genres: string[];
}
interface Store {
  searchBy: string;
  sortBy: string;
  movie: any;
}
export const ResultBody = () => {
  const { movies, isFetching } = useSelector<
    Store,
    { movies: Record<string, any>; isFetching: boolean }
  >(state => ({
    movies: state.movie.get('movies'),
    isFetching: state.movie.get('isFetching'),
  }));

  const dispatch = useDispatch();
  const yetLoader = (
    condition: boolean,
    then: React.ReactNode,
    otherwise: React.ReactNode,
  ): React.ReactNode => (condition ? then : otherwise);

  const resultClass = classNames('content', {
    'content-not-found': movies.length === 0,
    'content-films-list': movies.length > 0,
  });
  const router = useRouter();
  const { searchValue } = router.query;

  useEffect(() => {
    if (searchValue !== 'undefined') {
      dispatch(getMoviesBySearchQuery(searchValue));
    }
  }, []);

  return (
    <div className={resultClass}>
      {movies.length === 0
        ? yetLoader(
            isFetching,
            <Preloader />,
            <NoFilmsFound className="content-not-found__text" />,
          )
        : movies.map(
            ({ id, title, poster_path, release_date, genres }: MoviesTypes) => (
              <MovieItem
                key={id}
                id={id}
                handleOnClick={() => dispatch(getMovieDetailsById(id))}
                title={title}
                src={poster_path}
                year={release_date}
                genre={genres}
              />
            ),
          )}
    </div>
  );
};
