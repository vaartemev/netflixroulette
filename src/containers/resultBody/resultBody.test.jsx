import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { ResultBody } from './resultBody';
import { configureStore } from '../../store';

const store = configureStore();

describe('ResultBody', () => {
  const output = mount(
    <Router>
      <Provider store={store}>
        <ResultBody />
      </Provider>
    </Router>,
  );
  it('Render correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
