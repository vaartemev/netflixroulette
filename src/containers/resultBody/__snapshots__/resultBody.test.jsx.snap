// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultBody Render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <withRouter()>
        <Component
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <div
            className="content content-not-found"
          >
            <Preloader>
              <div
                className="preloader"
              >
                <div
                  className="loader"
                />
              </div>
            </Preloader>
          </div>
        </Component>
      </withRouter()>
    </Provider>
  </Router>
</BrowserRouter>
`;
